### PR TITLE
ui: Add launch in context /topology?filter=<G>;hightlight=<G>

### PR DESCRIPTION
**Pull request by hunchback**
- Open Skydive UI with custom view based on URL query string.
- Support for Multiple query keys,values

- Current support is for filter and highlight (gremlin expression)

Some examples:

http://localhost:8082/topology?filter=G.V().Has("Name", "AGGR1")

http://localhost:8082/topology?filter=G.V().Has("Name", "AGGR1")&highlight=G.V().Has("Name", Regex("AGGR1"))

http://localhost:8082/topology?filter=G.V().Has("Probe", Regex("fabric"))&highlight=G.V().Has("Name", Regex("AGGR1"))

Co-authored-by:    eranra <eranra@ERAN-TP>

### CODE REVIEW

**Comment by safchain** [comment]
run cdd-overview

File: http/server.go [hunk 1]
```diff
@@ -306,6 +306,7 @@ func NewServer(host string, serviceType common.ServiceType, addr string, port in
 	router.PathPrefix("/statics").HandlerFunc(server.serveStatics)
 	router.PathPrefix(ExtraAssetPrefix).HandlerFunc(server.serveStatics)
 	router.HandleFunc("/login", server.serveLogin)
+	router.PathPrefix("/topology").HandlerFunc(server.serveIndex)
```
Comment (safchain) [inline http/server.go]: I would prefer to not add an endpoint right now as not used by the backend

File: statics/js/components/topology.js [hunk 2]
```diff
@@ -285,6 +285,21 @@ var TopologyComponent = {
 
     this.setGremlinFavoritesFromConfig();
 
+    if (typeof(this.$route.query.highlight) !== "undefined") {
+        this.topologyEmphasize = this.$route.query.highlight;
```
Comment (safchain) [inline statics/js/components/topology.js]: nit: indentation,  2 extra spaces

File: statics/js/components/topology.js [hunk 3]
```diff
@@ -285,6 +285,21 @@ var TopologyComponent = {
...
+        setTimeout(function(){self.emphasizeGremlinExpr();}, 2000)
+    }
+
+    if (typeof(this.$route.query.filter) !== "undefined") {
+        this.topologyFilter = this.$route.query.filter;
```
Comment (safchain) [inline statics/js/components/topology.js]: same here

File: http/server.go (hunk 1 above)
Comment (hunchback) [reply http/server.go]: I added this as else working in VueRouter(mode: 'history') does not work as explained:

(https://router.vuejs.org/en/essentials/history-mode.html)[history-mode]

File: statics/js/components/topology.js (hunk 2 above)
Comment (hunchback) [reply statics/js/components/topology.js]: ok

File: statics/js/components/topology.js (hunk 3 above)
Comment (hunchback) [reply statics/js/components/topology.js]: ok

File: statics/js/components/topology.js [hunk 4]
```diff
@@ -285,6 +285,21 @@ var TopologyComponent = {
 
     this.setGremlinFavoritesFromConfig();
 
+    if (typeof(this.$route.query.highlight) !== "undefined") {
+      this.topologyEmphasize = this.$route.query.highlight;
+      setTimeout(function(){self.emphasizeGremlinExpr();}, 2000)
```
Comment (safchain) [inline statics/js/components/topology.js]: I don't think the timeout is needed as WebSocket event will trigger the function ? By the way I didn't have it working, It triggers exceptions, please check the console to see if it the same for you

Comment (eranra) [reply statics/js/components/topology.js]: - Hi Sylvain, when I wrote the original code I didn't had the timeout in place, and was thinking (like you) that there is no need, later I discovered that it just doesn't work with time-out., I checked 1 second and it worked, but decide on 2 seconds just to be on the safe side (also, it looks visually nice to see the nodes, and than the highlight appears).

- With regards to the exception, can you share ... I am running a version that is 99% similar to this one (only change is in the PathPrefix("/topology") that should make a difference) and I don't get any exception ....

Comment (safchain) [reply statics/js/components/topology.js]: I really think the setTimeout is useless but that's fine to let it I guess, here the exception I get, without or with 

```
topology.js:452 Uncaught TypeError: Cannot read property 'indexOf' of null
    at a.endsWith (topology.js:452)
    at a.n [as endsWith] (vue.min-2.1.9.js:6)
    at a.emphasizeGremlinExpr (topology.js:595)
    at a.n [as emphasizeGremlinExpr] (vue.min-2.1.9.js:6)
    at later (utils.js:8)
```

Comment (safchain) [reply statics/js/components/topology.js]: The issue is not due to your patch, can you fix it with this change the line https://github.com/skydive-project/skydive/blob/master/statics/js/components/topology.js#L463

by this one 
```
if (favorite) self.topologyEmphasize = favorite;
```

BTW a ";" is missing here at the end of the setTimeout line

Comment (hunchback) [reply statics/js/components/topology.js]: But we already protect against `favorite` being undefined here: https://github.com/skydive-project/skydive/blob/master/statics/js/components/topology.js#L455

Plus such a fix (if needed) this outside the context of the current PR. (so can be addressed separately).

Comment (safchain) [reply statics/js/components/topology.js]: Ok I will do

**Pull request merged**